File: march-2026/3-19-march-madness/march-madness.py
def upset_probability(matchups):
  results = []

  # for each matchup
  for matchup in matchups:
    # unpack the array for easier use
    team_a, seed_a, team_b, seed_b = matchup

    # use given formula :
    # upset_probability = higher_seed / (higher_seed + lower_seed)
    ups_probability = seed_a / (seed_a + seed_b)

    # add to results array and round to 2 decimal
    results.append(round(ups_probability, 2))

  return results

File: march-2026/3-19-march-madness/march_madness.js
// March Madness 🏀
// shercodes

function upsetProbability(matchups) {
  // Defines the array for upset.
  let upset = [];

  for (let i = 0; i < matchups.length; i++) {
    let match = matchups[i];
  
    // Determines the higher seed (seedA) and lower seed (seedB).
    const seedA = match[1];
    const seedB = match[3];
    
    // Calculates the upset probability for each matchup.
    const prob = seedA / (seedA + seedB);
    const result = Number(prob.toFixed(2));
    upset.push(result);
  }
  
  return upset;
};